libraries/google_cloud_scheduler_job.rb

```diff
@@ -52,6 +54,8 @@ def parse
     @description = @fetched['description']
     @schedule = @fetched['schedule']
     @time_zone = @fetched['timeZone']
+    @state = @fetched['state']
+    @paused = @fetched['paused']
```

Comment:
here paused is an enum value under state.
so when you will try to access @fetched['paused'] it will fail.

Please see the response of the api below:


{
  "name": string,
  "description": string,
  "schedule": string,
  "timeZone": string,
  "userUpdateTime": string,
  **"state": enum ([State]**(https://cloud.google.com/scheduler/docs/reference/rest/v1beta1/projects.locations.jobs#State)),
  "status": {
    object ([Status](https://cloud.google.com/scheduler/docs/reference/rest/Shared.Types/Status))
  },
  "scheduleTime": string,
  "lastAttemptTime": string,
  "retryConfig": {
    object ([RetryConfig](https://cloud.google.com/scheduler/docs/reference/rest/v1beta1/projects.locations.jobs#RetryConfig))
  },
  "attemptDeadline": string,
  "legacyAppEngineCron": boolean,

  // Union field target can be only one of the following:
  "pubsubTarget": {
    object ([PubsubTarget](https://cloud.google.com/scheduler/docs/reference/rest/v1beta1/projects.locations.jobs#PubsubTarget))
  },
  "appEngineHttpTarget": {
    object ([AppEngineHttpTarget](https://cloud.google.com/scheduler/docs/reference/rest/v1beta1/projects.locations.jobs#AppEngineHttpTarget))
  },
  "httpTarget": {
    object ([HttpTarget](https://cloud.google.com/scheduler/docs/reference/rest/v1beta1/projects.locations.jobs#HttpTarget))
  }
  // End of list of possible types for union field target.
}
api - 
[response api](https://cloud.google.com/scheduler/docs/reference/rest/v1beta1/projects.locations.jobs#resource:-job)

state enum has below values:
[Enum values](https://cloud.google.com/scheduler/docs/reference/rest/v1beta1/projects.locations.jobs#state)

Reply:
Got it. So if I pull out the stuff related to `paused` completely, this should work because we'll be able to test if `state` is equal to `PAUSED` (in the case of a paused job), right?

Reply:
Or, since `paused` is a top level attribute in Terraform itself, should the interface here stay the same, but we should just populate it from `@fetched['state']`?

Reply:
If this is something we should implement by making an adjustment in inspec's fork of magic-modules, happy to work on trying to implement it that way too (I didn't know it existed at the time I created this).

Reply:
Yes! That is correct.

Reply:
Yes You can contribute to Magic module that will be the right thing to do.